src/Microsoft.DotNet.Build.Tasks/PackageFiles/RunnerTemplate.Unix.txt

```diff
@@ -95,7 +95,15 @@ if [ "$(uname -s)" == "Darwin" ]; then
 elif [ "$(uname -s)" == "Linux" ]; then
   # On Linux, we'll enable core file generation unconditionally, and if a dump
   # is generated, we will print some useful information from it and delete the
-  # dump immediately. 
+  # dump immediately.
+
+  if [ -e /proc/self/coredump_filter ]; then
```

Comment:
I didn't dig in but we also need to be careful about which tools we start taking a dependency on to capture these dumps. In my ideal world the dump capturing would be handled by the host or a wrapper around the host itself, as opposed to a bunch more code in these scripts.